modules/openapi-generator/src/main/java/org/openapitools/codegen/AbstractGenerator.java

```diff
@@ -36,21 +40,73 @@
 
 public abstract class AbstractGenerator {
     private static final Logger LOGGER = LoggerFactory.getLogger(AbstractGenerator.class);
+    
+    /**
+     * Is the minimal-file-update option enabled?
+     * @return Option value
+     */
+    public abstract boolean getEnableMinimalUpdate();
 
-    @SuppressWarnings("static-method")
+    /**
+     * Write String to a file, formatting as UTF-8
+     * @param filename The name of file to write
+     * @param contents The contents string.
+     * @return File representing the written file.
+     * @throws IOException If file cannot be written.
+     */
     public File writeToFile(String filename, String contents) throws IOException {
-        LOGGER.info("writing file " + filename);
-        File output = new File(filename);
+        return writeToFile(filename, contents.getBytes(Charset.forName("UTF-8")));
+    }
 
+    /**
+     * Write bytes to a file
+     * @param filename The name of file to write
```

Comment:
[nits] Please insert blank comment line between the description and the list of tags.